src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -2315,6 +2311,76 @@ public function get_test_file_uploads() {
 		return $result;
 	}
 
+	/**
+	 * Tests if any of the WordPress Core classes are overridden.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @return array
+	 */
+	public static function get_test_core_classes_paths() {
+		$overridden_classes = array();
+		/*
+		 * Bypass check for WP_Object_Cache.
+		 *
+		 * It is intended that persistent caching plugins replace the core version
+		 * of WP_Object_Cache with their own version so it shouldn't be reported
+		 * as an error in site health.
+		 */
+		$wp_core_classes = WP_Autoload::CLASSES_PATHS;
+		unset( $wp_core_classes['wp_object_cache'] );
+		foreach ( $wp_core_classes as $class_name => $class_path ) {
+			$default_path = ABSPATH . $class_path;
+
+			/*
+			 * Init a Reflection class to get the real path of the class.
+			 * This is performend inside an output buffer to avoid any errors
+			 * that might be thrown by the ReflectionClass constructor.
+			 */
+			ob_start();
+			$reflection = new \ReflectionClass( $class_name );
+			ob_end_clean();
+
+			if ( $default_path !== $reflection->getFileName() ) {
+				$overridden_classes[ $reflection->getName() ] = $reflection->getFileName();
+			}
+		}
+
+		$result = array(
+			'label'       => __( 'All WordPress Core classes can be properly loaded.' ),
+			'status'      => 'good',
+			'badge'       => array(
+				'label' => __( 'Security' ),
+				'color' => 'blue',
+			),
+			'description' => sprintf(
+				'<p>%s</p>',
+				__( 'Plugins and Themes should not override the default WordPress Core classes, as that might lead to unexpected side-effects on your site.' )
+			),
+			'actions'     => '',
+			'test'        => 'core_classes_paths',
+		);
+
+		if ( ! empty( $overridden_classes ) ) {
+			$result['status'] = 'critical';
+			$result['label']  = __( 'WordPress Core classes are being overridden' );
+			$result['description'] .= sprintf(
+				'<p>%s</p>',
+				__( 'The following WordPress Core classes are being overridden:' )
+			);
+
+			$result['description'] .= '<table class="widefat striped health-check-table">';
+			foreach ( $overridden_classes as $class_name => $class_path ) {
+				$result['description'] .= '<tr>';
```

Comment:
There doesn't seem to be a matching `</tr>`.

Reply:
Good catch! Added 👍